crates/router/src/services/api.rs

```diff
@@ -393,9 +397,23 @@ pub async fn call_connector_api(
     state: &AppState,
     request: Request,
 ) -> CustomResult<Result<types::Response, types::Response>, errors::ApiClientError> {
+    let url = reqwest::Url::parse(&request.url)
+        .into_report()
+        .change_context(errors::ApiClientError::UrlEncodingFailed)?;
+
+    let tag = router_env::opentelemetry::KeyValue {
+        key: "host".into(),
+        value: url.host_str().unwrap_or_default().to_string().into(),
+    };
+
     let current_time = Instant::now();
 
-    let response = send_request(state, request, None).await;
+    let response = metrics_request::record_operation_time(
+        send_request(state, request, None),
```

Comment:
Would it be better to put the time recording logic inside the function, so that wherever the function is used the time is recorded.